### PR TITLE
Improve hero fading animation in castle

### PR DESCRIPTION
**Pull request by vasilenkoalexey**
I propose another solution for the problem of troops selecting event during hero fade animation. We can just finish the animation without disabling the event. This is more in line with the original idea of interactive events in the castle without any freezes.

### CODE REVIEW

File: src/fheroes2/castle/castle_dialog.cpp [hunk 1]
```diff
@@ -322,328 +322,330 @@ int Castle::OpenDialog( bool readonly )
 
     // dialog menu loop
     while ( le.HandleEvents() ) {
-        // During hero purchase or building construction disable any interaction
-        if ( alphaHero >= 255 && fadeBuilding.IsFadeDone() ) {
```
Comment (ihhub) [inline src/fheroes2/castle/castle_dialog.cpp:326]: Hi @vasilenkoalexey , could you please highlight what changes are done except removing this check?

File: src/fheroes2/castle/castle_dialog.cpp [hunk 2]
```diff
@@ -322,328 +322,330 @@ int Castle::OpenDialog( bool readonly )
...
+        // selector troops event
+        if ( ( selectArmy2.isValid()
+               && ( ( le.MouseCursor( selectArmy1.GetArea() ) && selectArmy1.QueueEventProcessing( selectArmy2, &msg_status ) )
+                    || ( le.MouseCursor( selectArmy2.GetArea() ) && selectArmy2.QueueEventProcessing( selectArmy1, &msg_status ) ) ) )
+             || ( !selectArmy2.isValid() && le.MouseCursor( selectArmy1.GetArea() ) && selectArmy1.QueueEventProcessing( &msg_status ) ) ) {
+            // Just finish fade animation on selector troops event
+            if ( alphaHero != 255 ) {
+                alphaHero = 255;
+                fheroes2::Blit( surfaceHero, display, cur_pt.x, cur_pt.y + 356 );
             }
```
Comment (vasilenkoalexey) [inline src/fheroes2/castle/castle_dialog.cpp:355]: this part

**Comment by ihhub** [comment]
Hi @vasilenkoalexey , could you please fix merge conflicts here?

**Comment by ihhub** [comment]
Hi @Branikolog , would you mind to test these changes?

**Comment by Branikolog** [comment]
During fading animation any action with army makes hero and his troops appear immediately. But when we enter hero info window and then cancel it - fading animation continues.
If we enter Hero info window and change the position of army during fading animation - the army doesn't update in main castle window.
image
image

The current solution works not bad, Creatures appear immediately, when we click on army during fading animation, but it looks a little bit roughly... Is it possible not jump to final state, but speed up fading animation, so the selection would happen with small delay in about 0.5 seconds? I'm missing some smoothness in current solution. =)

**Comment by vasilenkoalexey** [comment]
> During fading animation any action with army makes hero and his troops appear immediately. But when we enter hero info window and then cancel it - fading animation continues.
> If we enter Hero info window and change the position of army during fading animation - the army doesn't update in main castle window.

Should be fixed now

**Comment by ihhub** [comment]
Hi @vasilenkoalexey , I tested these changes. I found one issue: once the construction of the building has started click on any troop and you will immediately notice that the animation suddenly jumps to the finished state what shouldn't happen.

**Comment by vasilenkoalexey** [comment]
> Hi @vasilenkoalexey , I tested these changes. I found one issue: once the construction of the building has started click on any troop and you will immediately notice that the animation suddenly jumps to the finished state what shouldn't happen.

Hi! We discussed the logic with the Branikolog and decided that the user can either wait for the animation to complete or immediately complete any fade animation with any left or right mouse click. Indeed, in this way we can perform any action with the mouse without any friezes, or enjoy the fading animation.

**Comment by ihhub** [comment]
Hi @vasilenkoalexey , can you please fix conflicts for this pull request? But before doing this I think we need to come up with a solution which is suitable for everyone as I can see that there are some disagreements about the way of animation. I can't tell the exact solution for now, to be honest :(

**Pull request closed without merging**